lib/Cake/Test/Case/Model/Datasource/Database/MysqlTest.php

```diff
@@ -389,6 +389,36 @@ public function testBuildColumn() {
 		$expected = '`testName`  CHARACTER SET utf8 COLLATE utf8_unicode_ci DEFAULT NULL';
 		$this->assertEquals($expected, $result);
 		$this->Dbo->columns = $restore;
+
+		$data = array(
+			'name' => 'testName',
+			'type' => 'integer',
+			'storage' => 1
+		);
+		$result = $this->Dbo->buildColumn($data);
+		$expected = '`testName` tinyint(4)';
+		$this->assertEquals($expected, $result);
+		$this->Dbo->columns = $restore;
+
+		$data = array(
+			'name' => 'testName',
+			'type' => 'integer',
+			'storage' => 2
```

Comment:
`length` is not an issue. Length was always confusing to MySQL users -- it has absolutely no relation to the storage requirement, it is for display purposes. CakePHP *does* make some assumptions with respect to `tinyint` that have a length of `1` to treat them as an abstract `boolean`, and I made sure this is still working (99% sure). If you had a `tinyint(1)` in your database then you would still end up with an abstract `boolean` in your schema. It's when you are using `tinyint(2)` (or `tinyint(3)`, etc, does not matter the length as long as it's not 1) that you are (still) getting an abstract `integer`. Before my change this `integer` would turn into a 4 bytes `int` in MySQL though. Now it turns back into a `tinyint` as expected, that is, one byte. 

The problem with `'storageType'='small'` is that you are still being too abstract. Who knows how many bytes you mean here, right? If somebody were to implement a new backend, they would still not know how many bytes they can safely use for that column (and not *too many* bytes).

Absolutely, I can rename it to `storageSize`.

In this implementation, if the combination of type and storage is not found, then it defaults to what was found before my change -- in that case, you get an `integer` (4 bytes). But I see what you mean for 6, because `'type' => 'integer', 'storageSize' => 6` should result in something like a `bigint`, right? My gut feeling is this is an "advanced" option, for people who know what they are doing and want tight control of storage because they have *a lot* of rows (we do) -- I would assume these people know that `bigint` exists. Maybe a `honorStorageSize` flag could be used on the datasource itself, and default to false, to bypass my code? Where would be the right place to specify such behavior?